packages/ruleset/src/functions/anchored-patterns.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * Copyright 2017 - 2024 IBM Corporation.
+ * SPDX-License-Identifier: Apache2.0
+ */
+
+const {
+  isStringSchema,
+  validateNestedSchemas,
+} = require('@ibm-cloud/openapi-ruleset-utilities');
+
+const { LoggerFactory } = require('../utils');
+
+let ruleId;
+let logger;
+module.exports = function (schema, _opts, context) {
+  if (!logger) {
+    ruleId = context.rule.name;
+    logger = LoggerFactory.getInstance().getLogger(ruleId);
+  }
+  return validateNestedSchemas(
```

Comment:
We may have talked about this but I double checked and believe the `validateSubschemas` function is a variation that will check every individual schema in the entire document. I think you can use that and be able to remove your `// Now check for "pattern" within allOf/anyOf/oneOf lists (if present).` block of code below since those schemas would be included.

Could simplify the function code a bit so I think it's worth trying!

Reply:
Switched to validateSubschemas() so that each element of an allOf/anyOf/oneOf is visited, but as we discussed, I left in the logic that looked for pattern fields within a schema's allOf/anyOf/oneOf lists.